src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/ThrowHelper.cs

```diff
@@ -83,6 +83,11 @@ internal static class ThrowHelper
         public static void ThrowInvalidOperationException_InvalidZeroByteRead() => throw CreateInvalidOperationException_InvalidZeroByteRead();
         [MethodImpl(MethodImplOptions.NoInlining)]
         public static Exception CreateInvalidOperationException_InvalidZeroByteRead() => new InvalidOperationException(SR.InvalidZeroByteRead);
+
+        [DoesNotReturn]
+        public static void ThrowNotImplementedException_UnflushedBytes() => throw CreateNotImplementedException_UnflushedBytes();
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static Exception CreateNotImplementedException_UnflushedBytes() => new NotImplementedException(SR.UnflushedBytesNotImplemented);
```

Comment:
```suggestion
        public static Exception CreateNotImplementedException_UnflushedBytes() => new NotSupportedException(SR.UnflushedBytesNotImplemented);
```

Reply:
It was noted in the approval as a NotImplementedException, 
```
namespace System.IO.Pipelines
{
    partial class PipeWriter 
    {
         public virtual bool CanGetUnflushedBytes => false;
         public virtual long UnflushedBytes => throw new NotImplementedException(some message here);
    }
}
```

Happy to change it if that's acceptable /cc @bartonjs

Reply:
I see.

Reply:
I think NotSupportedException would be more sensible, afterall it isn't that the base doesn't implement it it's that we don't support it which we already said by returning false from CanGetUnflushedBytes, but I follow specs when provided with them...

Reply:
NotSupportedException makes sense to me.  The comment from the API review was more "throw something, instead of returning 0/-1/etc".

Reply:
Imo this is something that any PipeWriter *could* reasonably support. So if this *is* throwing, it's almost certainly because it's just not implemented yet instead of it being some intentional decision. I don't feel too strongly though

Reply:
Changed to NotSupportedException and various names and resources updated to match. 
If I was looking at a log where someone had tried to use the new feature inappropriately seeing NotSupportedException sends me down a different investigative route than NotImplementedException.